medusa/providers/torrent/html/torrenting.py

```diff
@@ -0,0 +1,200 @@
+
+        # URLs
+        self.url = 'https://www.torrenting.com/'
+        self.urls = {
+            'login': urljoin(self.url, 'login.php'),
+            'search': urljoin(self.url, 'browse.php'),
+        }
+
+        # Proper Strings
+        self.proper_strings = ['PROPER', 'REPACK', 'REAL', 'RERIP']
+
+        # Miscellaneous Options
+        self.enable_cookies = True
+        self.cookies = ''
+
+        # Torrent Stats
+        self.minseed = None
+        self.minleech = None
+
+        # Cache
+        self.cache = tv.Cache(self)
+
+    def search(self, search_strings, age=0, ep_obj=None):
+        """
+        Search a provider and parse the results.
+
+        :param search_strings: A dict with mode (key) and the search value (value)
+        :param age: Not used
+        :param ep_obj: Not used
+        :returns: A list of search results (structure)
+        """
+        results = []
+        if not self.login():
+            return results
+
+        # Search Params
+        search_params = {
+            'c4': 1,  # TV/SD-x264
+            'c5': 1,  # TV/X264 HD
+            'c18': 1,  # TV/Packs
+            'c49': 1,  # x265 (HEVC)
+            'search': '',
+        }
+
+        for mode in search_strings:
+            log.debug('Search mode: {0}', mode)
+
+            for search_string in search_strings[mode]:
+
+                if mode != 'RSS':
+                    log.debug('Search string: {search}',
+                              {'search': search_string})
+                    search_params['search'] = search_string
+
+                response = self.session.get(self.urls['search'], params=search_params)
+                if not response or not response.text:
+                    log.debug('No data returned from provider')
+                    continue
+
+                results += self.parse(response.text, mode)
+
+        return results
+
+    def parse(self, data, mode):
+        """
+        Parse search results for items.
+
+        :param data: The raw response from a search
+        :param mode: The current mode used to search, e.g. RSS
+
+        :return: A list of items found
+        """
+        items = []
+
+        with BS4Parser(data, 'html5lib') as html:
+            torrent_table = html.find("table", {"id": "torrentsTable"})
+            if torrent_table:
+                torrent_rows = torrent_table.find_all('tr')
+
+                # Continue only if at least one release is found
+                if len(torrent_rows) < 1:
+                    log.debug('Data returned from provider does not contain any torrents')
+                    return items
+
+                # Skip column headers
+                for row in torrent_rows[1:]:
+                    try:
+                        torrent_items = row.find_all('td')
+                        title = torrent_items[1].find('a').get_text(strip=True)
+                        download_url = urljoin(self.url, torrent_items[2].find('a')['href'])
```

Comment:
If you use urljoin here you make the following check for download_url useless, as it will always be true